e2e/exhaustive_suite_test.go

```diff
@@ -85,52 +85,54 @@ func Exhaustive(t *testing.T) {
 			w := workload
 			d := deployer
 
-			t.Run(w.GetName(), func(t *testing.T) {
+			t.Run(w.GetName()+"-"+d.GetName(), func(t *testing.T) {
 				t.Parallel()
-				t.Run(d.GetName(), func(t *testing.T) {
-					t.Parallel()
-					testcontext.AddTestContext(t.Name(), w, d)
-					runTestFlow(t)
-					testcontext.DeleteTestContext(t.Name(), w, d)
-				})
+				runTestFlow(t, w, d)
 			})
 		}
 	}
 }
 
-func runTestFlow(t *testing.T) {
+func runTestFlow(t *testing.T, w workloads.Workload, d deployers.Deployer) {
 	t.Helper()
 
-	testCtx, err := testcontext.GetTestContext(t.Name())
-	if err != nil {
-		t.Fatal(err)
+	if !d.IsWorkloadSupported(w) {
+		t.Skipf("Workload %s not supported by deployer %s, skip test", w.GetName(), d.GetName())
 	}
 
-	if !testCtx.Deployer.IsWorkloadSupported(testCtx.Workload) {
-		t.Skipf("Workload %s not supported by deployer %s, skip test", testCtx.Workload.GetName(), testCtx.Deployer.GetName())
-	}
-
-	if !t.Run("Deploy", DeployAction) {
-		t.Fatal("Deploy failed")
-	}
+	t.Run("Deploy", func(t *testing.T) {
+		if err := d.Deploy(w); err != nil {
+			t.Fatal("Deploy failed")
+		}
```

Comment:
This does not improve the code. Being able to run a sub test defined elsewhere is nicer. It will make it easier to create more flows except this simple flow. I think #1598 is better in this case.